components/HTMLBlock/index.tsx

```diff
@@ -14,20 +14,38 @@ const extractScripts = (html: string = ''): [string, () => void] => {
 };
 
 interface Props {
-  children: React.ReactElement | string;
+  children?: React.ReactElement | string;
+  html?: string;
   runScripts?: boolean | string;
-  safeMode?: boolean;
+  safeMode?: boolean | string;
 }
 
-const HTMLBlock = ({ children = '', runScripts, safeMode = false }: Props) => {
-  if (typeof children !== 'string') {
-    throw new TypeError('HTMLBlock: children must be a string');
+const HTMLBlock = ({ children = '', html: htmlProp, runScripts, safeMode: safeModeRaw = false }: Props) => {
+  let html: string = '';
+  if (htmlProp !== undefined) {
+    // MDXish mode: use html prop from HAST properties
+    html = htmlProp;
+  } else if (typeof children === 'string') {
+    // MDX mode: children is a string
+    html = children;
+  } else {
+    // MDX mode: extract string from React children (text nodes)
+    const textContent = React.Children.toArray(children)
+      .map(child => (typeof child === 'string' ? child : ''))
+      .join('');
+    html = textContent;
```

Comment:
This used to throw a `TypeError('HTMLBlock: children must be a string')`. Is this new logic necessary for MDXish? If it is, we need to make sure this logic is backwards compatible, right?[^1] (Sorry if this feels nitpicky, but these types of open ended changes worry me! 😬)

[^1]: Maybe we could do that by pulling [the `isMdxish` var](https://github.com/readmeio/markdown/pull/1243/files?show-viewed-files=false&manifests=false#diff-fa3edc30a976965de6df1169b3fcbf44de455063f1213dd1e72637b0c48f64f0R40) up above this block and updating [the final `else`](https://github.com/readmeio/markdown/pull/1243/files?show-viewed-files=false&manifests=false#diff-fa3edc30a976965de6df1169b3fcbf44de455063f1213dd1e72637b0c48f64f0R31) to check that…